src/Storages/System/StorageSystemUserDefinedFunctions.cpp

```diff
@@ -0,0 +1,188 @@
+#include <Storages/System/StorageSystemUserDefinedFunctions.h>
+#include <DataTypes/DataTypeArray.h>
+#include <DataTypes/DataTypeDateTime.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/DataTypeEnum.h>
+#include <Columns/ColumnString.h>
+#include <Columns/ColumnArray.h>
+#include <Interpreters/Context.h>
+#include <Common/ExternalLoaderStatus.h>
+#include <Functions/UserDefined/ExternalUserDefinedExecutableFunctionsLoader.h>
+#include <Functions/UserDefined/UserDefinedExecutableFunction.h>
+#include <Processors/Sources/ShellCommandSource.h>
+#include <Common/Exception.h>
+
+
+namespace DB
+{
+
+StorageSystemUserDefinedFunctions::StorageSystemUserDefinedFunctions(
+    const StorageID & storage_id_, ColumnsDescription columns_description_)
+    : IStorageSystemOneBlock(storage_id_, std::move(columns_description_))
+{
+}
+
+ColumnsDescription StorageSystemUserDefinedFunctions::getColumnsDescription()
+{
+    return ColumnsDescription
+    {
+        // ===== System/Non-Config Fields (External Loader metadata) =====
+        {"name", std::make_shared<DataTypeString>(),
+            "Function name (UDF identifier)."},
```

Comment:
Just `UDF name`? (I couldn't find a mention of the term "UDF identifier" in our docs).